pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -396,6 +400,10 @@ def on_validation_batch_end(self, trainer, pl_module, outputs, batch, batch_idx,
             self._update(self.val_progress_bar_id, self.val_batch_idx, self.total_val_batches)
         self.refresh()
 
+    def on_validation_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
```

Comment:
Should we use epoch_end instead? TQDM does

Reply:
done!

Reply:
Because my observation was wrong.

TQDM:
- Update n on `on_eval_epoch_end`
- Update metrics on `on_eval_end`

So we can do the same for rich (as you did initially!)